lib/src/executables/react16_upgrade.dart

```diff
@@ -28,29 +30,84 @@ const _changesRequiredOutput = """
 Then, review the the changes, address any FIXMEs, and commit.
 """;
 
+YamlMap pubspecYaml;
+
+Map packagesToCheckFor = {
+  'react': ['^4.0.0', '^5.0.0'],
+  'over_react': ['^2.0.0', '^3.0.0']
+};
+
+getDependencyVersion(String packageName) {
+  if (pubspecYaml.containsKey('dependencies') &&
+      pubspecYaml['dependencies'].containsKey(packageName)) {
+    return VersionConstraint.parse(pubspecYaml['dependencies'][packageName]);
+  } else if (pubspecYaml.containsKey('dev_dependencies') &&
+      pubspecYaml['dev_dependencies'].containsKey(packageName)) {
+    return VersionConstraint.parse(
+        pubspecYaml['dev_dependencies'][packageName]);
+  }
+  return null;
+}
+
 void main(List<String> args) {
-  final query = FileQuery.dir(
-    pathFilter: isDartFile,
-    recursive: true,
-  );
-  exitCode = runInteractiveCodemodSequence(
-    query,
-    [
-      ReactDomRenderMigrator(),
-      ReactStyleMapsUpdater(),
-    ],
-    args: args,
-    defaultYes: true,
-    changesRequiredOutput: _changesRequiredOutput,
-  );
-
-  final logger = Logger('over_react_codemod.fixmes');
-  for (var dartFile in query.generateFilePaths()) {
-    final dartSource = File(dartFile).readAsStringSync();
-    if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
-      logger.severe(
-          'over_react_codemod validation comments are unaddressed in $dartFile');
-      exitCode = 1;
+  String constraintPackage;
+  VersionConstraint constraint;
+  final entrylogger = Logger('over_react_codemod.precheck');
+  try {
+    pubspecYaml = loadYaml(File('pubspec.yaml').readAsStringSync());
+  } catch (e) {
+    entrylogger.fine('This repo does not have a pubspec.yaml');
```

Comment:
#nit This error might not be entirely correct if `loadYaml` is what throws (due to a malformed pubspec).